### DISCUSSION TITLE
How can I yield and resume from a WASM context to the host?

### DISCUSSION THREAD

**Discussion by ghost** [bytecodealliance/wasm-micro-runtime#3245] [Q&A]
I am curious, is it possible to yield execution to the wasm host from the wasm code and resume it later?

**Comment by xujuntwt95329** [comment]
Currently there is no standard way to do this, no opcode can tell the runtime to save the state of current function and yield the execution. 
I just noticed [asyncify](https://kripken.github.io/blog/wasm/2019/07/16/asyncify.html) which is a binaryen pass to add `setlongjump`-like functionality to WebAssembly in the "user space", not sure if this is what you want.

**Comment by AjaniBilby** [comment]
wasmer's wasix standard does have support for `stack_checkpoint` and `stack_restore` which can allow you to emulate this behaviour purely in assembly, but wasmer has other trade offs, and I'm personally hoping iwasm enables this sort of behaviour soon-ish
